cmd/neofs-cli/internal/client/client.go

```diff
@@ -439,7 +440,7 @@ func PutObject(ctx context.Context, prm PutObjectPrm) (*PutObjectRes, error) {
 		buf := make([]byte, sz)
 
 		_, err = io.CopyBuffer(wrt, prm.rdr, buf)
-		if err != nil {
+		if err != nil && !errors.Is(err, io.EOF) {
```

Comment:
this obviously cannot be accepted as working fix w/o SDK doc changes at least. Lets wait for SDK changes and base on them